Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -18,37 +15,30 @@ def relocate_dynamic_linkage(relocation)
   def change_rpath(file, old_prefix, new_prefix)
     return if !file.elf? || !file.dynamic_elf?
 
-    patchelf = DevelopmentTools.locate "patchelf"
-    odie "Could not locate patchelf, please: brew install patchelf." if patchelf.nil?
-    cmd = [patchelf]
-
     old_rpath = file.rpath
+
     new_rpath = if old_rpath
       rpath = old_rpath.split(":")
                        .map { |x| x.sub(old_prefix, new_prefix) }
                        .select { |x| x.start_with?(new_prefix, "$ORIGIN") }
-
       lib_path = "#{new_prefix}/lib"
       rpath << lib_path unless rpath.include? lib_path
-      new_rpath = rpath.join(":")
-      cmd.push  "--force-rpath", "--set-rpath", new_rpath
 
-      new_rpath
+      rpath.join(":")
     end
 
     old_interpreter = file.interpreter
-    new_interpreter = if old_interpreter.nil?
-      nil
-    elsif File.readable? "#{new_prefix}/lib/ld.so"
+    new_interpreter = if File.readable? "#{new_prefix}/lib/ld.so"
       "#{new_prefix}/lib/ld.so"
     else
       old_interpreter.sub old_prefix, new_prefix
     end
-    cmd << "--set-interpreter" << new_interpreter if old_interpreter != new_interpreter
 
-    return if old_rpath == new_rpath && old_interpreter == new_interpreter
+    return if old_rpath == new_rpath && (old_interpreter == new_interpreter)
 
-    safe_system(*cmd, file)
+    file.rpath = new_rpath
+    file.interpreter = new_interpreter
+    file.save!
```

Comment:
Rather than using a member variable `@set`, instead factor out a function that returns `[new_interpreter, new-rpath]`, and then implement `relocate_dynamic_linkage_using_patchelf` and `relocate_dynamic_linkage_using_patchelf_rb`. The function `relocate_dynamic_linkage_using_patchelf` should be as similar as possible to the current existing logic.